app/models/order_detail.rb

```diff
@@ -716,16 +722,16 @@ def in_dispute?
   end
 
   def disputed?
+    # only used in specs
     dispute_at.present? && !canceled?
   end
 
-  def cancel_reservation(canceled_by, order_status: OrderStatus.canceled_status, admin: false, admin_with_cancel_fee: false)
-    res = reservation
-    res.canceled_by = canceled_by.id
+  def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderStatus.canceled_status, admin: false, admin_with_cancel_fee: false)
+    self.canceled_by = canceled_by.id
+    self.canceled_reason = canceled_reason
 
     if admin
-      res.canceled_at = Time.zone.now
-      return false unless res.save
+      return false unless update_attributes(canceled_at: Time.current)
```

Comment:
Just curious where this might fail? I'm wondering if we can just use `assign_attributes` and then the save gets handled further down. I care a little less about the double save from a performance standpoint, but more from the versions table being easier to read (it's not something I have to look at too ofter, but when I do it's good to have it sane).

However, for this, I would lean towards ease of implementation and readability.